### PR TITLE
Add documentation for spider identification

### PR DESCRIPTION
**Pull request by mschnitzer**
This adds documentation for the new spider identification process.

See the attached PDF file for the whole documentation.
[obs-admin-guide_color_en.pdf](https://github.com/openSUSE/obs-docu/files/591177/obs-admin-guide_color_en.pdf)

### CODE REVIEW

File: xml/obs_ag_spider_identification.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  "-//OASIS//DTD DocBook XML V4.5//EN"
+  "http://www.docbook.org/xml/4.5/docbookx.dtd"
+[
+  <!ENTITY % entities SYSTEM "entity-decl.ent">
+  %entities;
+]>
+<section id="_obs_spider_identification">
+ <title>Spider Identification</title>
+ <para>
+   OBS identifys spiders by using the gem <command>voight_kampff</command>. Spiders will be identified to optimize the
```
Comment (ChrisBr) [inline xml/obs_ag_spider_identification.xml]:
```
s/identifys/identifies/
```

Comment (ChrisBr) [inline xml/obs_ag_spider_identification.xml]: > Spiders will be identified to optimize the page content for them.

Actually I would say this is not true, it is even [forbidden](https://en.wikipedia.org/wiki/Cloaking)! It is more that we hide certain content / links for performance reasons.

File: xml/obs_ag_spider_identification.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+   page content for them.
+ </para>
+ <para>
+   To update the spider list, you must run the command <command>rake voight_kampf:import_user_agents</command>.
+   This downloads a list of spiders as a JSON file into the <filename>config/</filename> directory of the Rails application.
```
Comment (ChrisBr) [inline xml/obs_ag_spider_identification.xml]: > a list of spiders

I would phrase it like

> an updated list of user-agents

File: xml/obs_ag_spider_identification.xml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ </para>
+ <para>
+   Switch to the <filename>config/</filename> directory and open the <filename>crawler-user-agents.json</filename> file with
```
Comment (ChrisBr) [inline xml/obs_ag_spider_identification.xml]: Hmm, I'm missing here the part that this is only necessary if the bot is not included in the list / if it is a custom bot. And furthermore, I would prefer not to maintain a custom list for this and instead follow the recommended way and a send a PR to https://github.com/monperrus/crawler-user-agents. Like it is recommended in the [gem readme](https://github.com/biola/Voight-Kampff).

Comment (mschnitzer) [inline xml/obs_ag_spider_identification.xml]: I agree to the first part but I don't think that it should matter us whether the user is sending a PR or not. This guide explains how to add a new spider and not how to contribute to a 3rd-party-project. I mean, not everyone who is using OBS knows what git is (neither GitHub and Pull Requests).

File: xml/obs_ag_spider_identification.xml (hunk 1 above)
Comment (hennevogel) [inline xml/obs_ag_spider_identification.xml]: Please also do not mention 'gems' in user documentation.

File: xml/obs_ag_spider_identification.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  %entities;
+]>
+<section id="_obs_spider_identification">
+ <title>Spider Identification</title>
+ <para>
+   OBS identifys spiders by using the gem <command>voight_kampff</command>. Spiders will be identified to optimize the
+   page content for them.
+ </para>
+ <para>
+   To update the spider list, you must run the command <command>rake voight_kampf:import_user_agents</command>.
```
Comment (hennevogel) [inline xml/obs_ag_spider_identification.xml]: How about instead of duplicating the `voight_kampf` documentation you point to it? :-)

Comment (mschnitzer) [inline xml/obs_ag_spider_identification.xml]: I cannot comment the first comment of you - for some reason. Anyways, I agree to your first comment.

To the second comment: There is no real documentation for this gem. The documentation links either pointing to the GitHub repository or to rubydoc.org what's just a collection of methods and classes in that gem. I would not like to point the admin to the README.md of this Git repository, for several reasons:
* The README.md describes it without any examples and in 3-4 sentences like "run xyz", "there is your config", "edit it", "regex is possible".
* The README.md can always change
* This guide describes it more concrete
* We don't know if the GitHub repository will get moved to some other platforms like GitBucket.

I asked one of the documentation team and she says that it's maybe better to just keep the documentation as it is for the reasons I posted above.

**Comment by mschnitzer** [comment]
@ChrisBr @hennevogel please review my new commit. It'll get squashed afterwards.

File: xml/obs_ag_spider_identification.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  "-//OASIS//DTD DocBook XML V4.5//EN"
+  "http://www.docbook.org/xml/4.5/docbookx.dtd"
+[
+  <!ENTITY % entities SYSTEM "entity-decl.ent">
+  %entities;
+]>
+<section id="_obs_spider_identification">
+ <title>Spider Identification</title>
+ <para>
+   OBS identifies spiders by using an extension called <command>voight_kampff</command>. Spiders will be identified to hide certain
```
Comment (hennevogel) [inline xml/obs_ag_spider_identification.xml]: Nobody cares what this is called or how it is implemented, that is what I meant with my previous comment. Please describe the functionality, the OBS functionality. Something like:

OBS is hiding specific parts/pages of the application from search crawlers (duckduckgo, google etc.), mostly for performance reasons. Which user-agent strings are identified as crawlers is configured in the file
`<filename>/srv/www/obs/api/config/crawler-user-agents.json</filename>`

etc...

File: xml/obs_ag_spider_identification.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+   content and links for performance reasons.
+ </para>
+ <para>
+   To update the spider list, you must run the command <command>rake voight_kampf:import_user_agents</command> in the
```
Comment (hennevogel) [inline xml/obs_ag_spider_identification.xml]: bundle exec is missing

File: xml/obs_ag_spider_identification.xml (hunk 5 above)
Comment (bgeuken) [inline xml/obs_ag_spider_identification.xml]: Update in #40

**Pull request closed without merging**